Update toscalarfunction.md (#1535)
MicrosoftDocs/dataexplorer-docs: data-explorer/kusto/query/toscalarfunction.md

> [!TIP]
> You can use a [let statement](letstatement.md) for readability of the query when using `toscalar()`.

## Limitations

`toscalar()` can be calculated a constant number of times during the query execution which means that the function can't be applied on row-level (for-each-row scenario).
Usually, when this limitation is hit, the following error will be reported: `can't use '<column name>' as it is defined outside its row-context scope.`

> for example in this case, it will fail with the error `'toscalar': can't use 'x' as it is defined outside its row-context scope.` :

Reviewer: I get the following error: 

 Semantic error: Unresolved reference binding: 'x'

Author: because there is an ongoing PR that should improve the error message :)
It was not merged yet as I need to generate a short link to this section and add it to the error message

Reviewer: Ok great. Let me review the text first.

> for example in this case, it will fail with the error `'toscalar': can't use 'x' as it is defined outside its row-context scope.` :

Reviewer: ```suggestion
In the following example, the query fails with the error:

> `'toscalar': can't use 'x' as it is defined outside its row-context scope.` 
```

> [!TIP]
> You can use a [let statement](letstatement.md) for readability of the query when using `toscalar()`.

## Limitations

> `toscalar()` can be calculated a constant number of times during the query execution which means that the function can't be applied on row-level (for-each-row scenario).
> Usually, when this limitation is hit, the following error will be reported: `can't use '<column name>' as it is defined outside its row-context scope.`

Reviewer: ```suggestion
`toscalar()` can't be applied on a scenario that applies the function on each row. This is because the function can only be calculated a constant number of times during the query execution.
Usually, when this limitation is hit, the following error will be returned: `can't use '<column name>' as it is defined outside its row-context scope.`
```
